lib/libc/newlib/libc-hooks.c

```diff
@@ -270,3 +268,147 @@ __weak int *__errno(void)
 {
 	return z_errno();
 }
+
+
+#if !defined(_RETARGETABLE_LOCKING) || \
+	CONFIG_NEWLIB_LIBC_DYNAMIC_LOCK_MEM_SIZE == 0
```

Comment:
what happens if this block isn't compiled? (i.e. someone set the dynamic lock mem size to 0)

Reply:
> How do you think about about the other switch to reserve memory for the locks? Is it ok, to have one? Would you set the default value to 0? Actually, I dont like to set it to 0 as only the malloc_lock hooks work in this szenario, but I was concerned that existing applications could run out of memory in case I choose a too little default value